ui/Dockerfile

```diff
@@ -1,33 +1,66 @@
+ARG DOCKER_REGISTRY=ghcr.io
 ARG TOOLS_UV_VERSION=latest
-ARG TOOLS_NODE_VERSION=20
-FROM --platform=$BUILDPLATFORM node:${TOOLS_NODE_VERSION}-bookworm AS ui-builder
+ARG PROXY
+ARG NOPROXY
+
+### STAGE 1: base image
+FROM --platform=$BUILDPLATFORM $DOCKER_REGISTRY/astral-sh/uv:${TOOLS_UV_VERSION}-bookworm-slim AS base-os
```

Comment:
what's the reason we're using the astral-sh/uv image for the UI instead of the node image?